### PR TITLE
feat(pcc.user): add possibility to delete an active directory

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pcc-federation` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-10029 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I added the possibility to delete an active directory

### CODE REVIEW

**Review by anooparveti — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/federation-delete.controller.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+export default class ActiveDirectoriesDeleteController {
+  /* @ngInject */
+  constructor($translate, DedicatedCloud) {
+    this.$translate = $translate;
+    this.DedicatedCloud = DedicatedCloud;
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/federation-delete.controller.js:5]: it's a good practice to name the class instance with minus first letter.
```suggestion
    this.dedicatedCloud = DedicatedCloud;
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/federation-delete.controller.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  close() {
+    if (this.task.state === 'done') {
+      this.loaders.cancelModal = true;
+      this.goBack(false, null, true);
+    } else {
+      this.goBack();
+    }
+  }
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/federation-delete.controller.js:45]:
```suggestion
    if (this.task.state === 'done') {
      this.loaders.cancelModal = true;
      return this.goBack(false, null, true);
    } 
    
    return this.goBack();
  }
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/index.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import angular from 'angular';
+import 'angular-translate';
+
+import component from './federation-delete.component';
+import taskPollProgress from '../../../taskPollProgress';
+
+const moduleName = 'ovhManagerDedicatedCloudUsersFederationDelete';
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/index.js:7]:
```suggestion
const moduleName = 'ovhManagerDedicatedCloudUsersActiveDirectoryDelete';
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/index.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+angular
+  .module(moduleName, [taskPollProgress])
+  .component('dedicatedCloudUsersFederationDelete', component)
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/index.js:11]:
```suggestion
  .component('dedicatedCloudUsersActiveDirectoryDelete', component)
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/federation-delete.module.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import angular from 'angular';
+
+import routing from './federation-delete.routes';
+import federationDeleteComponent from '../../../../components/dedicated-cloud/users/activeDirectories/delete';
+
+const moduleName = 'dedicatedCloudUsersFederationDeleteModule';
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/federation-delete.module.js:6]:
```suggestion
const moduleName = 'dedicatedCloudUsersActiveDirectoriesDeleteModule';
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/federation-delete.routes.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('app.dedicatedCloud.details.users.federation-delete', {
+    url: '/federation/delete',
+    params: {
+      activeDirectory: null,
+    },
+    views: {
+      modal: {
+        component: 'dedicatedCloudUsersFederationDelete',
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/federation-delete.routes.js:9]:
```suggestion
        component: 'dedicatedCloudUsersActiveDirectoriesDeleteFederation',
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/index.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import angular from 'angular';
+import '@uirouter/angularjs';
+import 'oclazyload';
+
+const moduleName = 'dedicatedCloudUsersFederationDeleteLazyloading';
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/index.js:5]:
```suggestion
const moduleName = 'dedicatedCloudUsersActiveDirectoriesDeleteLazyloading';
```

File: packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/delete/federation-delete.routes.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('app.managedBaremetal.details.users.federation-delete', {
+    url: '/federation/delete',
+    params: {
+      activeDirectory: null,
+    },
+    views: {
+      modal: {
+        component: 'dedicatedCloudUsersFederationDelete',
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/delete/federation-delete.routes.js:9]:
```suggestion
        component: 'dedicatedCloudUsersActiveDirectoriesFederationDelete',
```

File: packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/delete/federation-delete.module.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import angular from 'angular';
+
+import routing from './federation-delete.routes';
+import federationDeleteComponent from '../../../../components/dedicated-cloud/users/activeDirectories/delete';
+
+const moduleName = 'managedBaremetalUsersFederationDeleteModule';
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/delete/federation-delete.module.js:6]: as we discuss you have to update module name according path of module.

File: packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/delete/index.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import angular from 'angular';
+import '@uirouter/angularjs';
+import 'oclazyload';
+
+const moduleName = 'managedBaremetalUsersFederationDeleteLazyloading';
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/delete/index.js:5]: same as here :
https://github.com/ovh/manager/pull/8107/files#r986777114

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/federation-delete.controller.js (hunk 2 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/federation-delete.controller.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/index.js (hunk 4 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/index.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/federation-delete.module.js (hunk 5 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/federation-delete.module.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/federation-delete.routes.js (hunk 6 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/federation-delete.routes.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/index.js (hunk 7 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/delete/index.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/delete/federation-delete.routes.js (hunk 8 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/delete/federation-delete.routes.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/delete/index.js (hunk 10 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/delete/index.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/index.js (hunk 3 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/delete/index.js]: It's applied.

**Review by mohammed-zahaf — APPROVED** [review]

**Pull request merged**